src/parallel_progress.jl

```diff
@@ -0,0 +1,376 @@
+ - `auto_reset_timer`: tinit in progresses will be reset at first call
+
+use p[i] to access the i-th progressmeter, and p[0] to access the main one
+
+# Example
+```jldoctest
+julia> using Distributed
+julia> addprocs(2)
+julia> @everywhere using ProgressMeter
+julia> p = MultipleProgress(fill(10,5); desc="global ", kws=[(desc="task \$i ",) for i in 1:5])
+       pmap(1:5) do x
+           for i in 1:10
+               sleep(rand())
+               next!(p[x])
+           end
+           sleep(0.01)
+           myid()
+       end
+```
+"""
+function MultipleProgress(progresses::AbstractVector{<:AbstractProgress},
+                          mainprogress::AbstractProgress;
+                          enabled = true,
+                          auto_close = true,
+                          count_finishes = false,
+                          count_overshoot = false,
+                          auto_reset_timer = true)
+    !enabled && return MultipleProgress(FakeChannel(), length(progresses))
+
+    channel = RemoteChannel(() -> Channel{NTuple{4,Any}}(1024))
+    mp = MultipleProgress(channel, length(progresses))
+    @async runMultipleProgress(progresses, mainprogress, mp;
+        auto_close=auto_close,
+        count_finishes=count_finishes, 
+        count_overshoot=count_overshoot, 
+        auto_reset_timer=auto_reset_timer)
+    return mp
+end
+
+"""
+    MultipleProgress(progresses::AbstractVector{Progress}; count_finishes=false, kwmain=(), kw...)
+
+is equivalent to:
+
+    MultipleProgress(progresses, Progress(main_length; kwmain...); count_finishes, kw...)
+"""
+function MultipleProgress(progresses::AbstractVector{Progress}; 
+                          count_finishes=false, kwmain=(), kw...)
+    main_length = count_finishes ? length(progresses) : sum(p->p.n, progresses)
+    mainprogress = Progress(main_length; kwmain...)
+    return MultipleProgress(progresses, mainprogress; count_finishes=count_finishes, kw...)
+end
+
+"""
+    MultipleProgress(progresses::AbstractVector{<:AbstractProgress}; count_finishes=false, kwmain=(), kw...)
+
+is equivalent to:
+
+    MultipleProgress(progresses, prog; count_finishes, kw...)
+
+with `prog` a `Progress` of the same length as `progresses` if `count_finishes == true`, 
+otherwise, `prog` is a `ProgressUnknown`
+"""
+function MultipleProgress(progresses::AbstractVector{<:AbstractProgress}; 
+                          count_finishes=false, kwmain=(), kw...)
+    if count_finishes
+        MultipleProgress(progresses, Progress(length(progresses); kwmain...); count_finishes=count_finishes, kw...)
+    else
+        MultipleProgress(progresses, ProgressUnknown(;kwmain...); count_finishes=count_finishes, kw...)
+    end
+end
+
+"""
+    MultipleProgress(mainprogress::AbstractProgress; auto_close=false, kw...)
+
+is equivalent to
+
+    MultipleProgress(AbstractProgress[], mainprogress; auto_close, kw...)
+"""
+function MultipleProgress(mainprogress::AbstractProgress; auto_close=false, kw...)
```

Comment:
Should `mainprogress` default to `ProgressUnknown` in this case?

Reply:
yes, I've added this